### PR TITLE
fix: prevent server crash and opaque errors on tool failure

### PR DESCRIPTION
**Pull request by baruchiro**
## Summary

This PR hardens error handling throughout the MCP server to prevent crashes from unhandled rejections and to eliminate "[object Object]" leaks when tools or the Actual API reject with plain objects instead of Error instances.

## Key Changes

- **New `normalizeError()` utility** (`src/utils/response.ts`): Extracts usable error messages from any thrown value (Error instances, plain objects, strings, null/undefined). Falls back to JSON serialization for unknown shapes instead of producing "[object Object]".

- **Enhanced `error()` and `errorFromCatch()` functions** (`src/utils/response.ts`):
  - Accept optional `ErrorContext` with `toolName` and `code` for structured error responses
  - Include tool name and error code in response text and `_meta` field for client-side auditing
  - `errorFromCatch()` now uses `normalizeError()` to handle non-Error rejections gracefully

- **Improved tool dispatcher** (`src/tools/index.ts`):
  - Pass `toolName` and `code: 'unknown_tool'` to error responses for unknown tools
  - Await tool handler execution to ensure rejections are caught and the finally block doesn't shut down the API prematurely
  - Pass `toolName` context to `errorFromCatch()` for better error attribution

- **Process-level error handlers** (`src/index.ts`):
  - Added `unhandledRejection` and `uncaughtException` listeners to prevent crashes from unhandled errors
  - Both log errors and allow the process to continue running

- **Comprehensive test coverage**:
  - `src/utils/response.test.ts`: Tests for all response utilities including edge cases (circular objects, null/undefined, plain objects with various field names)
  - `src/tools/index.test.ts`: Integration tests for tool dispatcher error handling, including verification that "[object Object]" is never leaked and that API shutdown happens after handler completion

## Implementation Details

- `normalizeError()` checks for `message`, `error`, `reason`, `description`, and `detail` fields on plain objects, preferring the first non-empty string found
- Error code is extracted from both Error instances (via `.code` property) and plain objects, and is lifted into the response `_meta` for structured error tracking
- Tool handler execution is now awaited to ensure proper error propagation and cleanup ordering
- Process-level handlers use a new `toErrorMessage()` helper (added to `src/index.ts`) for consistent error logging

https://claude.ai/code/session_01C7vMwu4EjFNauKe4JY8ZGa

## Summary by CodeRabbit

* **Bug Fixes**
  * Improved error handling to prevent silent application crashes from unhandled rejections and exceptions.
  * Enhanced error messages to include tool-specific information and error codes for better debugging.

* **Tests**
  * Added comprehensive error-handling test coverage.

[Review Change Stack](https://app.coderabbit.ai/change-stack/baruchiro/actual-mcp/pull/25?utm_source=github_walkthrough&utm_medium=github&utm_campaign=change_stack)

### CODE REVIEW

File: src/tools/index.test.ts
Comment (baruchiro) [inline src/tools/index.test.ts:1]: I feel this test file is too complicated, remove it.

File: src/tools/index.ts [hunk 1]
```diff
@@ -79,20 +79,24 @@ export const setupTools = (server: Server, enableWrite: boolean): void => {
...
+      const tool = allTools.find((t) => t.schema.name === toolName);
       if (!tool) {
-        return error(`Unknown tool ${name}`);
+        return error(`Unknown tool ${toolName}`, { toolName, code: 'unknown_tool' });
       }
 
+      // Reason: must `await` so any rejection from the handler is caught here
+      // instead of becoming an unhandled rejection (which would crash the
+      // process) and so the `finally` block doesn't shut down the API
+      // before the handler completes.
```
Comment (baruchiro) [inline src/tools/index.ts]: remove comment.
